common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1121,10 +1121,12 @@ public boolean matches(String expr) {
      */
     @Override
     public int hashCode() {
-        if (hash == 0) {
-            hash = PlatformDependent.hashCodeAscii(value, offset, length);
+        int h = hash;
```

Comment:
It's fix for a non-benign data race. See [1], [2].

[1] https://shipilev.net/blog/2016/close-encounters-of-jmm-kind/#wishful-benign-is-resilient
[2] http://openjdk.5641.n7.nabble.com/RFR-8166842-String-hashCode-has-a-non-benign-data-race-td286991.html